From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kishan Mochi <kishan.mochi@intel.com>
Date: Tue, 30 Nov 2021 21:47:40 +0530
Subject: [PATCH] Enable zram swap for device having <= 4G ram

zram swap to increase available memory when device is
under memory pressure.

Tracked-On: OAM-100202
Signed-off-by: Kishan Mochi <kishan.mochi@intel.com>
---
 groups/swap/zram_auto/check_lowmem.sh | 10 +++++-----
 groups/swap/zram_auto/init.rc         |  4 ++--
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/groups/swap/zram_auto/check_lowmem.sh b/groups/swap/zram_auto/check_lowmem.sh
index 5ae5b88..e938f86 100755
--- a/groups/swap/zram_auto/check_lowmem.sh
+++ b/groups/swap/zram_auto/check_lowmem.sh
@@ -1,13 +1,13 @@
 #!/vendor/bin/sh
 
-# 2G size in kB
-SIZE_2G=2097152
+# 4G size in kB
+SIZE_4G=4194304
 
 mem_size=`cat /proc/meminfo | grep MemTotal | tr -s ' ' | cut -d ' ' -f 2`
 
-if [ "$mem_size" -le "$SIZE_2G" ]
+if [ "$mem_size" -le "$SIZE_4G" ]
 then
-    setprop vendor.low_ram 1
+    setprop vendor.le_4g_ram 1
 else
-    setprop vendor.low_ram 0
+    setprop vendor.le_4g_ram 0
 fi
diff --git a/groups/swap/zram_auto/init.rc b/groups/swap/zram_auto/init.rc
index 0ace57e..3203833 100644
--- a/groups/swap/zram_auto/init.rc
+++ b/groups/swap/zram_auto/init.rc
@@ -4,7 +4,7 @@ on boot
     write /proc/sys/vm/page-cluster 0
 {{#swappiness}}
    # Avoid evicting pages and use zram disk
-    write /proc/sys/vm/swappiness 100
+    write /proc/sys/vm/swappiness 10
 {{/swappiness}}
 {{#disk_based_swap}}
    # Enable disk_based_swap on Chromium kernels
@@ -14,6 +14,6 @@ on boot
 on post-fs
     exec - system system -- /vendor/bin/check_lowmem.sh
 
-on property:vendor.low_ram=1
+on property:vendor.le_4g_ram=1
     # Enable swaps described in the fstab
     swapon_all /vendor/etc/fstab.${ro.hardware}
-- 
2.17.1

